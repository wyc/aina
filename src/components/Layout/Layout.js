import PropTypes from 'prop-types';
import React from 'react';

import Header from '../Header/Header';
import Footer from '../Footer/Footer';

const Layout = ({ children }) => {
  return (
    <div className='Layout'>
      <Header />

      <main className='Main'>{children}</main>

      <Footer />
    </div>
  );
};

Layout.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Layout;
