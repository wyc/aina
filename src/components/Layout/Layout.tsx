import React, { ReactElement, ReactNode } from 'react';

import Footer from '../Footer/Footer';
import Header from '../Header/Header';

export interface ILayoutProps {
  children: ReactNode;
}

const Layout = ({ children }: ILayoutProps): ReactElement => {
  return (
    <div className='Layout'>
      <Header />

      <main className='Main'>{children}</main>

      <Footer />
    </div>
  );
};

export default Layout;
