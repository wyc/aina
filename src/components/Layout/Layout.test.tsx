import React from 'react';
import renderer from 'react-test-renderer';

import Layout from './Layout';

describe('Layout', () => {
  it('renders correctly', () => {
    const props = { children: [] };
    const component = renderer.create(<Layout {...props} />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
