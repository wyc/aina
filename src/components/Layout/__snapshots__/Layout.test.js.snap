// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<div
  className="Layout"
>
  <header
    className="Header"
  >
    <a
      aria-current="page"
      className=""
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      style={Object {}}
    >
      Home
    </a>
  </header>
  <main
    className="Main"
  />
  <footer
    className="Footer"
  >
    <a
      aria-current="page"
      className=""
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      style={Object {}}
    >
      Home
    </a>
    <div
      className="Footer__copyright"
    >
      Â© 2019 Victoria H. Chang
    </div>
  </footer>
</div>
`;
